bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ThingTypeRegistry.java

```diff
@@ -161,6 +162,21 @@ public ChannelType getChannelType(Channel channel, Locale locale) {
         return null;
     }
 
+    /**
+     * Returns the state description for a given channel, if provided by a {@link DynamicStateDescriptionProvider}
+     *
+     * @param channel channel
+     * @param locale locale (can be null)
+     * @return state description or null if none provided for this channel
+     */
+    public StateDescription getStateDescription(Channel channel, Locale locale) {
```

Comment:
Please remove this method because you do not need to go through the `ThingTypeRegistry` but you can call your `DynamicStateDescriptionRegistry` directly.